R/calculate_age.R

```diff
@@ -0,0 +1,90 @@
+#' Calculates the end of each age group given a series of start values for each age group
+#'
+#' Creates a new column in the input data.table called 'age_end'.
+#'
+#' 'age_start' is inclusive while 'age_end' exclusive. So for example age group
+#' \eqn{a} with 'age_start' equal to 5 and 'age_end' equal to 10 spans
+#' \eqn{5 <= a < 10}.
+#'
+#' This function assumes that only the most detailed age groups are in the input
+#' dataset.
+#'
+#' @param dt data.table with age-specific data.
+#' * must contain columns specified in `id_cols`.
+#' * must include a numeric column called 'age_start' that contains no missing
+#' values.
+#' * each combination of `id_cols` must uniquely identify each row.
+#' @param id_cols character vector of id columns that uniquely identify each row
+#' of `dt`.
+#' * must include 'age_start'.
+#' @param terminal_age_end the numeric 'age_end' value for the terminal age
+#' group.
+#'
+#' @return modifies `dt` in place by creating a new numeric column called
+#' 'age_end'.
+#'
+#' @export
+#'
+#' @examples
+#' input_dt <- data.table::data.table(location = "France", year = 2010,
+#'                                    sex = "female",
+#'                                    age_start = 0:95,
+#'                                    value1 = 1, value2 = 2)
+#' id_cols <- c("location", "year", "sex", "age_start")
+#' value_cols <- c("value1", "value2")
+#' calculate_age_end(input_dt, id_cols, terminal_age_end = 125)
+calculate_age_end <- function(dt, id_cols, terminal_age_end = 125L) {
```

Comment:
what does the L do? And is that required? (Is 110 acceptable as opposed to 110L?)

Reply:
It makes it an integer type rather than numeric.